src/unraphael/dash/pages/4_compare.py

```diff
@@ -120,6 +125,44 @@ def align_images_widget(*, base_image: ImageType, images: list[ImageType]) -> li
     return res
 
 
+def display_two(base_image, images):
+    col1, col2 = st.columns(2)
+    
+    if 'count' not in st.session_state:
+        st.session_state.count = 0
+
+    if 'images' not in st.session_state:
+        st.session_state.images = images  # Directly assign the list
+
+    def display_image():
+        try:
+            image = st.session_state.images[st.session_state.count].data
+            col1.image(base_image, caption="Base Image", use_column_width=True)
+            col2.image(image, caption=f"Image {st.session_state.count + 1}", use_column_width=True)
+        except IndexError as e:
+            st.error(f"Error displaying image: {e}")
+
+    def next_image():
+        if st.session_state.count + 1 >= len(st.session_state.images):
+            st.session_state.count = 0
+        else:
+            st.session_state.count += 1
+
+    def previous_image():
+        if st.session_state.count > 0:
+            st.session_state.count -= 1
```

Comment:
If you want to make this cycle, I suggest something like this, also see comment above.

```suggestion
        st.session_state.count = (st.session_state.count - 1) // len(st.session_state.images)
```

Reply:
this does not seem to work; one cannot proceed or go back anymore when using 
st.session_state.count = (st.session_state.count - 1) // len(st.session_state.images)

Reply:
See above

Reply:
likewise, works!